### ISSUE TITLE
Error matching when characters like `®` are present in the text

### ISSUE THREAD

**Issue by TibsAtWork** [jdkato/txtlint#410]
## Summary
Characters like `®` or `™` (U+00AE and U+2122) seem to cause match offset calculations to go wrong.

For instance:
```
$ vale --output=line 'World Health Organization (WHO) (R) and WHO or WHO'
```
but
```
$ vale --output=line 'World Health Organization (WHO) ® and WHO or WHO'
stdin.txt:1:28:Test.WHO_example:'WHO' has no definition
stdin.txt:1:39:Test.WHO_example:'WHO' has no definition
```

## Details

I'm working on macOS 12.1 (Monterey) on an M1 machine.

I've tested against vale versions 2.11.2, 2.14.0 and a build of v2 (which reports v2.15.1). See the "**Exploring further**" section below for some debugging I did using that last.

With my `.vale.ini` as:
```
StylesPath = "styles"

[*]
BasedOnStyles = Vale, Test
```
and minimal setup
```
$ tree styles/
styles/
└─Test/
  └─WHO_example.yml
```
with `styles/Test/WHO_example.yml` as:
```
extends: conditional
message: "'%s' has no definition"
level: error
scope: text
ignorecase: false

first: '\bWHO\b'
second: '(?:\b[A-Z][a-z]+ )+\((WHO)\)'
```

I get:
```
$ vale --output=line 'World Health Organization (WHO) (R) and WHO or WHO'
```
which is a success (if I don't use `--output=line` it reports no errors/warnings), but
```
$ vale --output=line 'World Health Organization (WHO) ® and WHO or WHO'
stdin.txt:1:28:Test.WHO_example:'WHO' has no definition
stdin.txt:1:39:Test.WHO_example:'WHO' has no definition
```

On my mac, I entered the `®` at the command line using `OPT`-`R`, which matches the suggestion at https://en.wikipedia.org/wiki/Registered_trademark_symbol.

To double check, I've also done the same by putting the texts into files (`simple-who-good.txt` and `simple-who-bad.txt`) and run `vale simple-who-good.txt` and `vale simple-who-bad.txt`, and that gives the same results.

I did do a search of the vale repository to see if there were any references to telling vale that text is unicode, but didn't find anything.

## Exploring further

I cloned https://github.com/errata-ai/vale.git and edited `internal/check/conditional.go` to add `"fmt"` to the `import` block and some diagnostic printing (I'm new to go, so there may have been a better way to do this):
``` go
     77        matches := c.patterns[0].FindAllStringSubmatch(txt, -1)
     78        for _, mat := range matches {
     79                if len(mat) > 1 {
     80                        // If we find one, we store it in a slice associated with this
     81                        // particular file.
     82                        for _, m := range mat[1:] {
     83                                if len(m) > 0 {
     84                                        fmt.Printf("second: %v -> %#v\n", c.patterns[0], m)    // DEBUG
     85                                        f.Sequences = append(f.Sequences, m)
     86                                }
     87                        }
     88                }
     89        }
     90
     91        // Now we look for the antecedent.
     92        locs := c.patterns[1].FindAllStringIndex(txt, -1)
     93        if len(locs) > 0 {    // DEBUG
     94                 fmt.Printf("locs %v\n", locs)
     95        }                 
     96        for _, loc := range locs {
     97                 s := txt[loc[0]:loc[1]]
     98                 fmt.Printf("first: %v -> %#v | %#v\n", c.patterns[1], loc, s)    // DEBUG
     99                 if !core.StringInSlice(s, f.Sequences) && !isMatch(c.exceptRe, s) {
    100                         // If we've found one (e.g., "WHO") and we haven't marked it as
    101                         // being defined previously, send an Alert.
    102                         alerts = append(alerts, makeAlert(c.Definition, loc, txt))
    103                }
    104       }
```

and with that I see:
```
$ ~/sw/errata.ai/vale/bin/vale --output=line 'World Health Organization (WHO) (R) and WHO or WHO'
second: (?:\b[A-Z][a-z]+ )+\((WHO)\) -> "WHO"
locs [[27 30] [40 43] [47 50]]
first: \bWHO\b -> []int{27, 30} | "WHO"
first: \bWHO\b -> []int{40, 43} | "WHO"
first: \bWHO\b -> []int{47, 50} | "WHO"
```
which is what I would expect, but:
```
$ ~/sw/errata.ai/vale/bin/vale --output=line 'World Health Organization (WHO) ® and WHO or WHO'
second: (?:\b[A-Z][a-z]+ )+\((WHO)\) -> "WHO"
locs [[27 30] [38 41] [45 48]]
first: \bWHO\b -> []int{27, 30} | "WHO"
first: \bWHO\b -> []int{38, 41} | " WH"
first: \bWHO\b -> []int{45, 48} | " WH"
stdin.txt:1:28:Test.WHO_example:'WHO' has no definition
stdin.txt:1:39:Test.WHO_example:'WHO' has no definition
```
where the offset of the `first` match appears to be wrong.

Trying with `™` is subtly different, presumably because its UTF-8 representation is different:
```
$ ~/sw/errata.ai/vale/bin/vale --output=line 'World Health Organization (WHO) ™ and WHO or WHO'
second: (?:\b[A-Z][a-z]+ )+\((WHO)\) -> "WHO"
locs [[27 30] [38 41] [45 48]]
first: \bWHO\b -> []int{27, 30} | "WHO"
first: \bWHO\b -> []int{38, 41} | "d W"
first: \bWHO\b -> []int{45, 48} | "r W"
stdin.txt:1:28:Test.WHO_example:'WHO' has no definition
stdin.txt:1:39:Test.WHO_example:'WHO' has no definition
```
